### PR TITLE
Fix dns end point

### PR DESCRIPTION
**Pull request by Madanagopal19**
This PR fixes the dns endpoint issue 

**Which issue(s) this PR fixes** (optional)
Closes #DS-9811

Jenkins logs - https://jenkins.pwx.dev.purestorage.com/job/PDS/job/pds2.0-byoc-branch-build-test/856/console

### CODE REVIEW

File: drivers/unifiedPlatform/pdsLibs/dsLib.go [hunk 1]
```diff
@@ -211,3 +211,112 @@ func DeleteAllDeployments(projectId string) error {
 	log.Infof("Total number of deployments Deleted [%d]", numberOfDeploymentsDeleted)
 	return nil
 }
+
+// ParseInterfaceAndGetDetails takes interface as input and checks for the particular type and extracts the host and port information
+// Returns the host and port as dnsEndpoints
+func ParseInterfaceAndGetDetails(connectionDetails interface{}, dataServiceName string) (string, error) {
```
[the remark this answers is not in the document]
Comment (Madanagopal19) [reply drivers/unifiedPlatform/pdsLibs/dsLib.go:217]: this is specific to pds

File: tests/unifiedPlatform/pds2/pds_restore_test.go [hunk 2]
```diff
@@ -403,17 +403,17 @@ var _ = Describe("{UpgradeDataServiceImageAndScaleUpDsWithBackUpRestore}", func(
...
-					err := SetSourceKubeConfig()
-					log.FailOnError(err, "failed to switch context to source cluster")
-				}()
-				CheckforClusterSwitch()
-				_, err := WorkflowPDSRestore.CreateRestore(restoreName, latestBackupUid, restoreNamespace, PDS_DEFAULT_NAMESPACE)
-				log.FailOnError(err, "Restore Failed")
-				log.Infof("All restores - [%+v]", WorkflowPDSRestore.Restores)
-				log.Infof("Restore Created Name - [%s], UID - [%s]", *WorkflowPDSRestore.Restores[restoreName].Meta.Name, *WorkflowPDSRestore.Restores[restoreName].Meta.Uid)
-			})
+			//Step("Create Restore from the latest backup Id", func() {
```
[the remark this answers is not in the document]
Comment (Madanagopal19) [reply tests/unifiedPlatform/pds2/pds_restore_test.go]: done

File: drivers/unifiedPlatform/pdsLibs/dsLib.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Madanagopal19) [reply drivers/unifiedPlatform/pdsLibs/dsLib.go]: done

File: drivers/unifiedPlatform/stworkflows/pds/workflowDataservice.go [hunk 3]
```diff
@@ -223,16 +226,16 @@ func (wfDataService *WorkflowDataService) DeleteDeployment(deploymentId string)
 	return nil
 }
 
-func (wfDataService *WorkflowDataService) ValidateDNSEndpoint(deploymentId string) error {
+func (wfDataService *WorkflowDataService) ValidateDNSEndpoint(deploymentId, dataServiceName string) error {
```
[the remark this answers is not in the document]
Comment (Madanagopal19) [reply drivers/unifiedPlatform/stworkflows/pds/workflowDataservice.go]: done

**Review by atrivedi-px — COMMENTED** [review]
LGTM

**Review by atrivedi-px — APPROVED** [review]

**Review by aravindranath14 — APPROVED** [review]
LGTM

**Pull request merged**